test/e2e/framework/framework.go

```diff
@@ -138,6 +138,12 @@ func New(kubeConfigPath string) (*Framework, cleanUpFunc, error) {
 		return nil, nil, err
 	}
 
+	// Setup debug log level in ThanosQuery.
+	err = f.setupDebug()
```

Comment:
The setting won't be permanent because other tests override the CMO configuration (`TestPrometheusVolumeClaim` for instance). Is it really needed?

Reply:
I can move it closer to where I would like to see debug logs in this case where we query watchdog alert, but yes true it would be overridden.

Not sure I understand what you mean by "Is it really needed?"?

Reply:
I meant to ask if we always need to run with debug log level. I guess it's for troubleshooting when the test fails?

Reply:
Yes, exactly for that test only. I haven't checked if they are run in parallel as to one to override this configmap? If not will just create configmap in this test and wait for thanos query pod to be rolled out. My only fear is, this might affect the test, as restarting the pods is not the same as happens on a regular cluster and we might not catch addr resolve issues as it would not be running for long enough. 🤔 Any ideas/suggestions welcome!

Reply:
yeah, not sure what is best... Running always with debug log level might hide issues (I've seen so many bugs that don't manifest when you enable debug logs).

Reply:
Okay will remove the debug enabling, if we see the failures more often we can add it later.